src/NServiceBus.Core/Routing/RoutingSettings.cs

```diff
@@ -25,6 +27,38 @@ public void RouteToEndpoint(Type messageType, string destination)
         {
             Settings.GetOrCreate<UnicastRoutingTable>().RouteToEndpoint(messageType, destination);
         }
+
+        /// <summary>
+        /// Adds a static unicast route for all types contained in the specified assembly.
+        /// </summary>
+        /// <param name="assembly">The assembly whose messages should be routed.</param>
+        /// <param name="destination">Destination endpoint.</param>
+        public void RouteToEndpoint(Assembly assembly, string destination)
+        {
+            var routingTable = Settings.GetOrCreate<UnicastRoutingTable>();
+            foreach (var type in assembly.GetTypes())
+            {
+                routingTable.RouteToEndpoint(type, destination);
+            }
+        }
+
+        /// <summary>
+        /// Adds a static unicast route for all types contained in the specified assembly within the given namespace.
+        /// </summary>
+        /// <param name="assembly">The assembly whose messages should be routed.</param>
+        /// <param name="messageNamespace">The namespace of the messages which should be routed.</param>
+        /// <param name="destination">Destination endpoint.</param>
+        public void RouteToEndpoint(Assembly assembly, string messageNamespace, string destination)
+        {
+            // empty namespace is null, not string.empty
+            messageNamespace = messageNamespace == string.Empty ? null : messageNamespace;
+
+            var routingTable = Settings.GetOrCreate<UnicastRoutingTable>();
+            foreach (var type in assembly.GetTypes().Where(t => t.Namespace == messageNamespace))
```

Comment:
Discussed with @timbussmann will address this in another PR. We are going in with this first.